content/blog/improved-kubernetes-await-logic/index.md

```diff
@@ -0,0 +1,293 @@
+
+   For example this resource expects its "phase" status field to have a value of "Running":
+
+        "pulumi.com/waitFor": "jsonpath={.status.phase}=Running"
+
+   If a value isn't provided, the resource is considered ready when any value exists at the given path.
+   This resource will wait until it has a web hook configured with a CA bundle:
+
+        "pulumi.com/waitFor": "jsonpath={.webhooks[*].clientConfig.caBundle}"
+
+2. A string prefixed with "condition=" followed by the type of the condition and an optional status.
+   This matches the behavior of `kubectl wait --for=condition=...` and will wait until the resource has a matching condition.
+   The expected status defaults to "True" if not specified.
+   For example this will wait until the resource has a "Synced=True" condition:
+
+       "pulumi.com/waitFor": "condition=Synced"
+
+   and this will wait until it has a "Reconciling=False" condition:
+
+       "pulumi.com/waitFor": "condition=Reconciling=False"
+
+3. Lastly, the `pulumi.com/waitFor` annotation also accepts a JSON-encoded string containing an array of one or more "jsonpath=" or "condition=" expressions.
+   For example, this resource will be considered ready when it simultaneously has a "Bar=True" condition and a populated "foo" field.
+
+       "pulumi.com/waitFor": '["jsonpath={.foo}", "condition=Bar"]'
+
+### Custom readiness example: cert-manager
+
+[cert-manager](http://cert-manager.io) is a popular X.509 certificate management solution with some installation nuances.
+If you follow the default [instructions](https://cert-manager.io/docs/installation/) with a goal of also creating a new Certificate as part of your Pulumi stack, you might end up with something like this:
+
+```typescript
+import * as kubernetes from "@pulumi/kubernetes";
+
+const install = new kubernetes.yaml.v2.ConfigFile("install",
+  {
+    file: "https://github.com/cert-manager/cert-manager/releases/download/v1.15.2/cert-manager.yaml",
+  },
+);
+
+const issuer = new kubernetes.apiextensions.CustomResource("issuer",
+  {
+    apiVersion: "cert-manager.io/v1",
+    kind: "ClusterIssuer",
+    metadata: {
+      name: "selfsigned-issuer",
+    },
+    spec: {
+      selfSigned: {},
+    },
+  },
+  { dependsOn: [install] }
+);
+
+new kubernetes.apiextensions.CustomResource("certificate", {
+  apiVersion: "cert-manager.io/v1",
+  kind: "Certificate",
+  metadata: {
+    name: "selfsigned-cert",
+  },
+  spec: {
+    isCA: true,
+    commonName: "my-ca",
+    secretName: "root-secret",
+    privateKey: {
+      algorithm: "ECDSA",
+      size: 256,
+    },
+    issuerRef: {
+      name: issuer.metadata.name,
+      kind: issuer.kind,
+      group: "cert-manager.io",
+    },
+  },
+});
+```
+
+This uses a [v2 ConfigFile](https://www.pulumi.com/blog/kubernetes-yaml-v2/) to `kubectl apply` the cert-manager custom resource definitions, and then it creates a self-signed Issuer and Certificate using those CRDs.
```

Comment:
nit: add CRD abbreviation as well

eg: `the cert-manager CustomResourceDefinition (CRD), and then it creates ...`